File: tests/test_helpers/TestHelper.java
class TestHelper {
    public static void NO_RESULT() {
        System.exit(%d);             // RESULTLESS_EXIT_CODE
    }
    public static void RESULT(int result) {
        System.exit(result + (%d));  // RESULT_EXIT_CODE_SHIFT
    }
    
    public static void PASS() {
        System.exit(%d);            // MIN_RESULT
    }
    public static void FAIL() {
        System.exit(%d);            // MAX_RESULT
    }
}

File: tests/test_helpers/test_helper.c
#include <stdlib.h>

#define NO_RESULT() exit({% RESULTLESS_EXIT_CODE %})
#define RESULT(r) exit(r+({% RESULT_EXIT_CODE_SHIFT %}))
#define PASS() exit({% MAX_RESULT %})
#define FAIL() exit({% MIN_RESULT %})